src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorDocumentOnTypeFormattingEndpoint.cs

```diff
@@ -81,6 +81,12 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(DocumentOnTypeFormatting
             return null;
         }
 
+        if (!_optionsMonitor.CurrentValue.FormatOnType)
```

Comment:
This is where we check the option on the razor server side

Reply:
We could potentially unify on the server side. @allisonchou thoughts?

Reply:
That would be ideal to me

Reply:
I'll do it as a followup, filed #8159  and assigned myself. Just because it will take more in depth testing with VS Code as well

Reply:
I agree unifying is a good idea. Don't think it should be blocker on this PR though